Prevent cancellation of a plan or domain on an AT site

jsnajdr opened the pull request:
Cancelling the Business plan or a domain on an AT site leaves that site in an inconsistent state: the site is on Pressable, but no longer has the Business plan or a primary domain other than *.wordpress.com. In such state, not even the hard-revert script works.

When trying to remove a plan or domain purchase from an AT site, I show this dialog:

<img width="506" alt="screen shot 2017-07-06 at 14 21 37" src="https://user-images.githubusercontent.com/664258/27911408-4294cd7a-6259-11e7-9533-e9099bbe647b.png">

I need help with the text on the dialog, of course. @iamtakashi, can you help? The idea is that the user needs to contact support if they want to cancel the plan or the domain. There is the "Need to chat" button, but I'm not sure if that's the best solution. Chat isn't always available, which means no button is rendered, and maybe the wording could be better.

I also need to test this on cancelling a domain for the site -- so far, I tested only the plan cancellation, as the domain is usually included in the plan and cannot be cancelled separately anyway.

What's a bit strange that I can't add multiple domains to an AT site. The "Add Domain" button is not rendered. It's [because the site has a `jetpack` property](https://github.com/Automattic/wp-calypso/blob/90dec34daa06fe18c9de52c207492596f7a7180d/client/my-sites/upgrades/domain-management/list/index.jsx#L230) set to `true` on an AT site. @lamosty, isn't that a bug? Or is there a reason why an AT site shouldn't have multiple alternate domain names?

iamtakashi commented:
> the user needs to contact support if they want to cancel the plan or the domain.

Am I correct to think that sending an email is the best way to contact our support in this matter when the chat isn't available?

jsnajdr commented:
> Am I correct to think that sending an email is the best way to contact our support with this matter when the chat isn't available?

Calypso has a dedicated page to contact support: https://wordpress.com/help/contact That's probably the one that should be used -- it checks what the user is eligible for and offers the right options: live chat, creating a ticket, public forum...

iamtakashi commented:
OK, then we could add a "Contact Support" button that take you to the page when the chat isn't available. Can we make the button to open the chat if it is available?

The heading "Remove WordPress,com Business" is a bit odd to be there. I don't think that's necessary to be there as the context is clear.

How about something like this? We should ask the Editorial for the copy though.

![image](https://user-images.githubusercontent.com/908665/27963544-52a45558-632d-11e7-9215-56654e09b13d.png)

lamosty commented:
@iamtakashi @jsnajdr  from the Slack convo we had, I concluded that we don't want to go this route and allow reverting sites without custom domains instead. Should we close this? Or am I missing something? :)

jsnajdr commented:
@lamosty See my comment on 284-gh-automated-transfer. As cancelling the domain is a guaranteed way to get a bricked site and needing to ask support for a hard-revert, I think it's still a good idea to prevent domain cancellations somehow.

Cancelling the Business plan is OK in any case though, that was a misunderstanding on my part.

jsnajdr commented:
I rebased and updated this PR -- we might need it to fix the business cancellation issue.

We need to update the copy of the dialog, and also provide a good way to contact support -- either just link to `/help/contact`, or be smarter and provide a Chat button in case chat is available.

apeatling commented:
@donpark Mind taking a look at this one and see if it's something we can merge?

apeatling commented:
@jhnstn This one would be good for the atomic team if you see fit. For now I am closing because it's not getting any activity.

donpark commented:
hmm. didn't get a Github notification.
anyway, noted this PR in the relevant card at AT Issues board.

jhnstn commented:
Re-opening since this canceling a Business plan for Atomic sites has quite a few hidden problems that don't happen if support handles the cancellation.

jhnstn commented:
@iamtakashi  I updated the copy and added the contact us button for when chat is not available. Also from your sketch it looks like you don't want the form heading. I removed it but I can add it back if that was not what you were looking for:

![screen shot 2017-12-22 at 1 41 15 pm](https://user-images.githubusercontent.com/744755/34314030-c03dd74e-e724-11e7-8ad1-240e994a5802.png)

michelleweber commented:
Let's nix "unfortunately." Let's lose the opening negativity, and make it a little more active:

To cancel your WordPress.com Business Plan, please contact our support team -- a Happiness Engineer will take care of it.

or

To cancel your WordPress.com Business Plan, please chat with our support team  -- a Happiness Engineer will take care of it..

rachelmcr commented:
@jhnstn Still working on this one?

iamtakashi commented:
@jhnstn Sorry for not responding sooner. I'm happy without a heading, and it looks fine.

jhnstn commented:
Here is the latest screenshot:

![screen shot 2018-01-24 at 3 51 27 pm](https://user-images.githubusercontent.com/744755/35363369-87d8a780-011e-11e8-8204-bb3ef2cb886d.png)

lancewillett commented:
Minor design note: the `--` should be a proper em-dash `—` in the text. With a space on either side.
